### PR TITLE
Add scale to ground speed for >= API 1.45

### PR DESCRIPTION
**Pull request by haslinghuis**
Fixes: #465

### CODE REVIEW

**Comment by klutvott123** [comment]
Did this change? Or was it always like that?

File: src/SCRIPTS/BF/PAGES/rescue.lua [hunk 1]
```diff
@@ -16,7 +16,12 @@ if apiVersion >= 1.41 then
     fields[#fields + 1] = { t = "Angle",            x = x,          y = inc.y(lineSpacing), sp = x + sp, min = 0, max = 200, vals = { 1, 2 } }
     fields[#fields + 1] = { t = "Initial Altitude", x = x,          y = inc.y(lineSpacing), sp = x + sp, min = 20, max = 100, vals = { 3, 4 } }
     fields[#fields + 1] = { t = "Descent Distance", x = x,          y = inc.y(lineSpacing), sp = x + sp, min = 30, max = 500, vals = { 5, 6 } }
-    fields[#fields + 1] = { t = "Ground Speed",     x = x,          y = inc.y(lineSpacing), sp = x + sp, min = 30, max = 3000, vals = { 7, 8 } }
+
+    if apiVersion >= 1.45 then
+        fields[#fields + 1] = { t = "Ground Speed", x = x,          y = inc.y(lineSpacing), sp = x + sp, min = 30, max = 3000, vals = { 7, 8 } }
+    else
+        fields[#fields + 1] = { t = "Ground Speed", x = x,          y = inc.y(lineSpacing), sp = x + sp, min = 0, max = 3000, vals = { 7, 8 }, scale = 100 }
+    end
```
Comment (klutvott123) [inline src/SCRIPTS/BF/PAGES/rescue.lua:24]: Did the min limit change? Hasn't it always been 0.3m/s? Are you sure you're scaling it for the right api version here?

**Comment by haslinghuis** [comment]
Yes scale seem to be wrong before (in any case looking at https://github.com/betaflight/betaflight/pull/11834/files
Seems to be changed in 1.45 (at leas the scale).

:confused: Might need to update the label:

{ "RETURN SPEED CM/S", OME_UINT16 | REBOOT_REQUIRED, NULL, &(OSD_UINT16_t){ &gpsRescueConfig_rescueGroundspeed, 0, 3000, 1 } },

File: src/SCRIPTS/BF/PAGES/rescue.lua (hunk 1 above)
Comment (haslinghuis) [reply src/SCRIPTS/BF/PAGES/rescue.lua:24]: Min has changed. See below.

Comment (klutvott123) [reply src/SCRIPTS/BF/PAGES/rescue.lua:24]: It's still 0.3m/s in the configurator, but sure, we can do 0. Just need to decide if it's supposed to be in m/s for api version >= 1.45 or below

**Comment by klutvott123** [comment]
> Yes scale seem to be wrong before (in any case looking at https://github.com/betaflight/betaflight/pull/11834/files Seems to be changed in 1.45 (at least the scale).
> 
> 😕 Might need to update the label:
> 
> { "RETURN SPEED CM/S", OME_UINT16 | REBOOT_REQUIRED, NULL, &(OSD_UINT16_t){ &gpsRescueConfig_rescueGroundspeed, 0, 3000, 1 } },

I think the label should be consistent with the configurator label, and the number itself should be shown as in the configurator.


This is the kind of bugs we get when labels aren't used. "needs coordination with betaflight configurator" lets us look at the milestone and pick up all the changes that have been made to user changeable parameters. Sure it might be that the person implementing the changes in firmware are doing the configurator PR as well, but still labels should be used.

**Comment by haslinghuis** [comment]
Configurator UI seems not have been updated ground speed. Last time it was updated: https://github.com/betaflight/betaflight-configurator/pull/1946

**Comment by haslinghuis** [comment]
@ctzsnooze - can you help us out here?

File: src/SCRIPTS/BF/PAGES/rescue.lua (hunk 1 above)
Comment (klutvott123) [reply src/SCRIPTS/BF/PAGES/rescue.lua:24]: @haslinghuis This has to be changed around. This adds the scaling for apiVersion < 1.45

Comment (haslinghuis) [reply src/SCRIPTS/BF/PAGES/rescue.lua]: Fixed :)

File: src/SCRIPTS/BF/PAGES/rescue.lua [hunk 2]
```diff
@@ -16,7 +16,12 @@ if apiVersion >= 1.41 then
     fields[#fields + 1] = { t = "Angle",            x = x,          y = inc.y(lineSpacing), sp = x + sp, min = 0, max = 200, vals = { 1, 2 } }
     fields[#fields + 1] = { t = "Initial Altitude", x = x,          y = inc.y(lineSpacing), sp = x + sp, min = 20, max = 100, vals = { 3, 4 } }
     fields[#fields + 1] = { t = "Descent Distance", x = x,          y = inc.y(lineSpacing), sp = x + sp, min = 30, max = 500, vals = { 5, 6 } }
-    fields[#fields + 1] = { t = "Ground Speed",     x = x,          y = inc.y(lineSpacing), sp = x + sp, min = 30, max = 3000, vals = { 7, 8 } }
+
+    if apiVersion >= 1.45 then
+        fields[#fields + 1] = { t = "Ground Speed", x = x,          y = inc.y(lineSpacing), sp = x + sp, min = 30, max = 3000, vals = { 7, 8 }, scale = 100 }
+    else
+        fields[#fields + 1] = { t = "Ground Speed", x = x,          y = inc.y(lineSpacing), sp = x + sp, min = 0, max = 3000, vals = { 7, 8 } }
```
Comment (klutvott123) [inline src/SCRIPTS/BF/PAGES/rescue.lua:23]: Shouldn't  `min` be switched around too?

**Review by klutvott123 — APPROVED** [review]

**Pull request merged**